src/input_handler.rs

```diff
@@ -34,62 +34,84 @@ impl InputHandler for Anodium {
                     true
                 } else if action != KeyAction::None {
                     self.shortcut_handler(action);
-                    self.input_state.keyboard.is_focused()
+                    self.input_state.borrow().keyboard.is_focused()
                 } else {
                     true
                 }
             }
             InputEvent::PointerMotion { event, .. } => {
-                self.input_state.pointer_location =
-                    self.clamp_coords(self.input_state.pointer_location + event.delta());
-                self.on_pointer_move(event.time());
-                self.surface_under(self.input_state.pointer_location)
-                    .is_none()
+                let pointer_location;
+                {
+                    let mut input_state = self.input_state.borrow_mut();
+                    input_state.pointer_location =
+                        self.clamp_coords(input_state.pointer_location + event.delta());
+                    pointer_location = input_state.pointer_location;
+                }
+
+                if let Some(region) = self.region_manager.region_under(pointer_location) {
+                    {
+                        let mut input_state = self.input_state.borrow_mut();
+                        input_state.pointer_location =
+                            self.clamp_coords(input_state.pointer_location + event.delta());
+                    }
```

Comment:
Actually, what is the reason for adding event.delta() to pointer_location twice here

Reply:
right, that doesn't make sense. I probably copied something to much when testing it. Will have a PR with removes that in a second.